src/jit/gentree.h

```diff
@@ -4209,6 +4215,42 @@ struct GenTreeSIMD : public GenTreeOp
 };
 #endif // FEATURE_SIMD
 
+#if defined(_TARGET_XARCH_) && !defined(LEGACY_BACKEND)
+struct GenTreeHWIntrinsic : public GenTreeOp
```

Comment:
They will not be exactly same. They have different types of intrinsic ID and compile flags. Meanwhile, `GenTreeHWIntrinsic` will have one more field to encode overload info because the new SIMD hardware intrinsics have much more complex overloads than the existing SIMD intrinsics that can impact instruction selection. I will add the overload info into the HWIntrinsic table when I implement a subset of SIMD hardware intrinsics.
So `GenTreeHWIntrinsic` could be a subclass of `GenTreeSIMD`, but I prefer to define two struct at this step. But if you think it is necessary to combine them now, I will change.

Reply:
I think it is desirable to combine them, if only to avoid duplication and extra cases where they can be in common. If the additional information makes it larger than a small node, then perhaps `GenTreeHWIntrinsic` should derive from `GenTreeJITIntrinsic` (or whatever).

Reply:
> then perhaps GenTreeHWIntrinsic should derive from GenTreeJITIntrinsic (or whatever).

I like this solution, but why does node size matter here?

Reply:
> why does node size matter here?

I don't think we want `GT_SIMD` nodes to become large, but if they share a type, and that type grows too large for a small node, they would all have to be large.

Reply:
I see, thank you for explaining.